### DISCUSSION TITLE
Recommended Knitr settings for R Markdown docs?

### DISCUSSION THREAD

**Discussion by kubu4** [RobertsLab/resources#1745] [Q&A]
Anyone willing to share some examples of their Knitr settings they like to use? Especially for R Markdown docs with heavy usage of Bash chunks (not sure if this impacts setting choices or not)?

**Comment by sr320** [comment]
Here is an example
https://github.com/course-fish546-2023/steven-coursework/blob/main/assignments/code/3.1-blast.Rmd

the key is for bash code your chunk should an r chunk, indicating bash engine  - example

````
```{r blast-look, engine='bash', eval=TRUE}
head -2 ../output/Ab_4-uniprot_blastx.tab
```
````

**Reply by sr320** [reply]
and this is how it renders https://rpubs.com/sr320/1026094

and few resources 
- https://sr320.github.io/course-fish546-2023/lectures/03-rstudio.html
- https://sr320.github.io/course-fish546-2023/assignments/03-knit.html

**Reply by kubu4** [reply]
Oooh! Thanks!

**Reply by kubu4** [reply]
It auto-numbers sections based on markdown headings (i.e. a `#` gets rendered as Section 1)? Interesting!

**Reply by kubu4** [reply]
I've added an additional parameter to the setup chunk:

` comment = ""         # Prevents appending '##' to beginning of lines in code output`

**Reply by kubu4** [reply]
When you have this in your chunk header, ` eval=TRUE`, what's your workflow like, @sr320? Do you run the chunk first to make sure it works and then have to do the Knitting after that - essentially running the chunk (and, the entire R Markdown file) twice? 

Seems time consuming, particularly for long-running tasks.

**Reply by sr320** [reply]
I do not think about knitting until end. I run each chunk and if there are any chunks I want to/ need to display i indicate 'eval=true'. Most of this are quick (plots and head). When I need something intense to be included in knit I 'cache=true'. This way chunk is not re-run every knit.

**Reply by kubu4** [reply]
To clarify this, do you replace `eval=TRUE` with `cache=TRUE`?

Basically, I'm having difficulty getting a compute-intense chunk to evaluate (basically, I'll like the run time to be displayed) _and_ have it cached, so I don't have to re-run it over and over when I tweak other chunks or add additional markdown to the other portions of the document.

**Reply by sr320** [reply]
> do you replace eval=TRUE with cache=TRUE?

Yes - that is correct. Note the first time you knit it will run, but after that it will used cache. 

Just curious- why run at all in knit? as opposed to just running chunk prior to knit.

**Reply by kubu4** [reply]
If I don't `eval=TRUE` while knitting, then the output from the chunk isn't displayed in the knitted output. Isn't that correct? I guess I'm confused on what's happening when knitting...

**Reply by sr320** [reply]
Correct, the chunk will not run when you knit. This would be key if generating a plot.

If you were just creating files in chunk, you could run a following chunk to display (eg, `cat`, `head`) which likely would take less time. Example is when I run HiSat, stringtie, convert SAMs, I do not `eval` but `eval` following chunk eg -https://github.com/urol-e5/deep-dive/blob/main/D-Apul/code/05.32-lncRNA-discovery-overview.md#2-stringtie

**Reply by kubu4** [reply]
Arlgihty, thanks. I think that helps me understand how it all works.

Maybe I just need to take a different approach to the long-running chunks (e.g. don't have additional code in those chunks that prints stuff).
